### PR TITLE
Revert "[NPM] Set `CsmEnabled` in agent payload from `runtime_security_config.enabled`"

### PR DESCRIPTION
**Pull request by lebauce**
Reverts DataDog/datadog-agent#29449

### CODE REVIEW

File: pkg/network/encoding/marshal/modeler_test.go [hunk 1]
```diff
@@ -1,67 +0,0 @@
...
-	model "github.com/DataDog/agent-payload/v5/process"
-	"github.com/stretchr/testify/assert"
-
-	"github.com/DataDog/datadog-agent/pkg/config/mock"
-	"github.com/DataDog/datadog-agent/pkg/network"
-)
-
-func TestConnectionModelerAgentConfiguration(t *testing.T) {
-	tests := []struct {
-		npm, usm, ccm, csm bool
```
Comment (hmahmood) [inline pkg/network/encoding/marshal/modeler_test.go:22]: Can we keep this test with `csm` removed?

**Review by brycekahle — APPROVED** [review]

File: pkg/network/encoding/marshal/modeler_test.go (hunk 1 above)
Comment (lebauce) [reply pkg/network/encoding/marshal/modeler_test.go:22]: :+1:

**Comment by leeavital** [comment]
Please make sure this doesn't break the backend build (I believe it will because we're removing a field that's referenced in BE code)

**Comment by hmahmood** [comment]
> Please make sure this doesn't break the backend build (I believe it will because we're removing a field that's referenced in BE code)

We are not removing the field (no changes to agent payload); just not setting it.

**Comment by hmahmood** [comment]
@lebauce are we merging this for 7.59? The only issue remaining for me is https://github.com/DataDog/datadog-agent/pull/29588#discussion_r1777333138

**Review by hmahmood — APPROVED** [review]

**Pull request closed without merging**